app/src/main/java/com/github/palFinderTeam/palfinder/map/MapsActivity.kt

```diff
@@ -0,0 +1,146 @@
+package com.github.palFinderTeam.palfinder.map
+
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.location.Location
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.core.app.ActivityCompat
+import com.github.palFinderTeam.palfinder.R
+
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.github.palFinderTeam.palfinder.databinding.ActivityMapsBinding
+import com.github.palFinderTeam.palfinder.meetups.MeetUp
+import com.github.palFinderTeam.palfinder.meetups.activities.MEETUP_SHOWN
+import com.github.palFinderTeam.palfinder.meetups.activities.MeetUpView
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.Marker
+import com.google.android.gms.maps.model.MarkerOptions
+import java.util.*
+
+class MapsActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnMarkerClickListener {
+
+    private lateinit var binding: ActivityMapsBinding
+    private lateinit var fusedLocationClient: FusedLocationProviderClient
+    private lateinit var lastLocation: Location
+    private lateinit var map: GoogleMap
+
+    companion object {
+        private const val USER_LOCATION_PERMISSION_REQUEST_CODE = 1
+        private var meetUps = LinkedList<MeetUp>()
+
+
+        /**
+         * add a marker corresponding to a meetup
+         */
+        fun addMeetUpMarker(meetUp: MeetUp) {
+            meetUps.add(meetUp)
+            }
+
+    }
+
+    // TODO add all meetup to map
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityMapsBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        val mapFragment = supportFragmentManager
+            .findFragmentById(R.id.map) as SupportMapFragment
+        mapFragment.getMapAsync(this)
+
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
+
+    }
+
+    private fun addMarkers(){
+        meetUps.forEach{
+            val position = LatLng(it.location.latitude, it.location.longitude)
+            val marker = map.addMarker(MarkerOptions().position(position).title(it.uuid))
+            if(marker != null) marker.tag = it
```

Comment:
you can replace it with a 'double-bang' operator (maker!!.tag), which return a NullPointerException if marker is null, or with a ?, which simply does nothing if marker is null